### PR TITLE
Feature/fix workflow rebazer

### PR DESCRIPTION
**Pull request by IgorAlbert**
Added some methods to try to work with the latest time of the repository, so the rebazer can see that there are some new changes.

### CODE REVIEW

**Comment by IgorAlbert** [comment]
improvements are welcome but not approve. 
getGitHubChecksTimeStamps in the class GithubConnector could throw a Exception.

File: src/test/java/org/retest/rebazer/connector/GithubConnectorTest.java [hunk 1]
```diff
@@ -139,10 +144,221 @@ void getAllPullRequests_should_return_all_pull_requests_as_list() throws Excepti
...
+
+		assertThat( cut.newestChecksTime( pullRequest ) ).isEqualTo( "2019-01-04T15:30:59Z" );
+		assertThat( cut.getLatestUpdate( pullRequest ).getLastUpdate() ).hasSameTimeAs( "2019-01-04T15:30:59Z" );
+
+	}
+
+	@Test
+	void newestChecksTime_should_return_newest_time_of_the_checks() {
+		final PullRequest pullRequest = mock( PullRequest.class );
+		final String checksFinished1 =
```
Comment (beatngu13) [inline src/test/java/org/retest/rebazer/connector/GithubConnectorTest.java]: To keep tests simple, one shouldn't test multiple aspects within a single test case. `checksFinished1`, `checksFinished2`, and `checksFinished3` look unrelated. If it doesn't make sense to extract separate test case as it is just different test data, have a look at [parameterized tests](https://baeldung.com/parameterized-tests-junit-5/). (Same goes for the other tests below.)

**Comment by beatngu13** [comment]
Looks good to me, what do you think @martin-v?

However, @IgorAlbert there are 7 test errors to be fixed: https://travis-ci.com/retest/rebazer/builds/108826032#L616

File: src/main/java/org/retest/rebazer/service/PullRequestLastUpdateStore.java [hunk 2]
```diff
@@ -10,25 +12,27 @@
 @Service
 public class PullRequestLastUpdateStore {
 
-	private final Map<RepositoryConfig, Map<Integer, String>> pullRequestUpdateStates = new HashMap<>();
+	private final Map<RepositoryConfig, Map<Integer, Date>> pullRequestUpdateStates = new HashMap<>();
+	public static final String DEFAULT_REPOSITORY_TIME = "2019-01-01T00:00:00Z";
```
Comment (martin-v) [inline src/main/java/org/retest/rebazer/service/PullRequestLastUpdateStore.java]: * Use Date object as constant
* Use FALLBACK as name instead of DEFAULT
* Use `new Date(0)` to create the object with the same value

File: src/main/java/org/retest/rebazer/connector/BitbucketConnector.java [hunk 3]
```diff
@@ -40,11 +42,12 @@ public BitbucketConnector( final RepositoryConfig repoConfig, final RestTemplate
 	@Override
 	public PullRequest getLatestUpdate( final PullRequest pullRequest ) {
 		final DocumentContext jsonPath = jsonPathForPath( requestPath( pullRequest ) );
+		final Date repositoryTime = Date.from( OffsetDateTime.parse( jsonPath.read( "$.updated_on" ) ).toInstant() );
```
Comment (martin-v) [inline src/main/java/org/retest/rebazer/connector/BitbucketConnector.java]: extract `Date.from( OffsetDateTime.parse( string ).toInstant() );` to method, it exists twice.

File: src/main/java/org/retest/rebazer/connector/GithubConnector.java [hunk 4]
```diff
@@ -37,11 +40,14 @@ public GithubConnector( final RepositoryConfig repoConfig, final RestTemplateBui
 	@Override
 	public PullRequest getLatestUpdate( final PullRequest pullRequest ) {
 		final DocumentContext jsonPath = jsonPathForPath( requestPath( pullRequest ) );
+		final String repositoryTimeAsString = jsonPath.read( "$.updated_at" );
+		final Date repositoryTime = Date.from( OffsetDateTime.parse( repositoryTimeAsString ).toInstant() );
+		final Date checksTime = Date.from( OffsetDateTime.parse( newestChecksTime( pullRequest ) ).toInstant() );
```
Comment (martin-v) [inline src/main/java/org/retest/rebazer/connector/GithubConnector.java]: Same here for `Date.from( OffsetDateTime.parse( string ).toInstant() );`

File: src/main/java/org/retest/rebazer/connector/GithubConnector.java [hunk 5]
```diff
@@ -127,6 +142,19 @@ private static String requestPath( final PullRequest pullRequest ) {
 		return JsonPath.parse( json.getBody() ).<List<String>> read( "$.check_runs[*].conclusion" );
 	}
 
+	private List<String> getGitHubChecksTimeStamps( final PullRequest pullRequest ) {
```
Comment (martin-v) [inline src/main/java/org/retest/rebazer/connector/GithubConnector.java]: This method dupplicates code from `getGitHubChecks`

File: src/main/java/org/retest/rebazer/service/PullRequestLastUpdateStore.java (hunk 2 above)
Comment (IgorAlbert) [reply src/main/java/org/retest/rebazer/service/PullRequestLastUpdateStore.java]: I used the suggested method to convert the string, because new date is deprecated and LocalDate makes things more cumbersome.
